src/core/border_router/routing_manager.cpp

```diff
@@ -2376,6 +2427,19 @@ void RoutingManager::Nat64PrefixManager::HandleDiscoverDone(const Ip6::Prefix &a
     }
 }
 
+RoutingManager::Nat64PrefixManagerState RoutingManager::Nat64PrefixManager::GetState(void)
+{
+    Nat64PrefixManagerState state = kNat64StateDisabled;
+
+    VerifyOrExit(IsEnabled());
+    VerifyOrExit(Get<RoutingManager>().IsRunning(), state = kNat64StateNotRunning);
+    VerifyOrExit(mPublishedPrefix.IsValidNat64(), state = kNat64StateIdle);
+    state = kNat64StateActive;
+
+exit:
+    return state;
```

Comment:
It may also work if we remember `mState` instead of `mEnabled`?
Thought?

Reply:
I would suggest maintaining `mEnabled` instead of `mState`. Since we only have a single place changing the value of `mEnabled`, while we will have 4 (start/stop/SetEnabled/Evaluate) for mState. And I think it makes it hard to varify the correctness of the returned state.

Reply:
Up to you. Both model can have their pros and cons.

With tracking `State`, we need to have `switch(mState)` at the top of all methods and decide what to do under each state (and transition to other state if we need to). Sometimes this helps make the code simpler.